lang/cs/Org.Apache.REEF.Tang.Tests/Injection/TestInjection.cs

```diff
@@ -304,6 +304,55 @@ public void TestExternalObject()
 
             Assert.IsNotNull(o.ExternalObject is ExternalClass);
         }
+        
+        /// <summary>
+        /// In this test, interface is generic of T. Implementations use different T as type. 
+        /// When doing injection, we need to get type of generic type T so that to get generic interface
+        /// </summary>
+        [TestMethod]
+        public void TestInjectionWithT()
+        {
+            var c = TangFactory.GetTang().NewConfigurationBuilder()
+                .BindImplementation(GenericType<IMyOperator<int>>.Class, GenericType<MyOperatorImpl<int>>.Class)
+                .BindImplementation(GenericType<IMyOperator<int[]>>.Class, GenericType<MyOperatorImpl<int[]>>.Class)
+                .BindImplementation(GenericType<IMyOperator<string>>.Class, GenericType<MyOperatorImpl<string>>.Class)
+                .BindNamedParameter(typeof(MessageType), typeof(int[]).AssemblyQualifiedName)
+                .Build();
+
+            var injector = TangFactory.GetTang().NewInjector(c);
+
+            var msgType = (string)injector.GetNamedInstance(typeof(MessageType));
```

Comment:
wouldn't it be possible to use the typed version of `GetNamedInstance` to avoid the typecast?